analyzers/src/SonarAnalyzer.Common/Common/AnalyzerConfiguration.cs

```diff
@@ -76,38 +81,35 @@ public bool IsEnabled(string ruleKey)
                     && enabledRules.Contains(ruleKey);
             }
 
-            public void Initialize(AnalyzerOptions options)
+            public void Initialize(AnalyzerOptions options, IRuleLoader ruleLoader)
             {
-                if (isInitialized)
+                var currentSonarLintXmlPath = GetSonarLintXmlPath(options);
+                if (isInitialized && loadedSonarLintXmlPath == currentSonarLintXmlPath)
```

Comment:
Aren't parallel builds are handled by different processes?

Reply:
I don't know. We've seen static state persisted state between analysis. Might worth to try to verify it locally with some logging.

Reply:
For a solution with multiple project this should not be a problem since we have the same SonarLint.xml configuration for all the projects. 

For building multiple solutions in parallel, I think we don't need to support this scenario. According to [our documentation](https://docs.sonarqube.org/latest/analysis/scan/sonarscanner-for-msbuild/):
```
Concurrent analyses (i.e. parallel analysis of two solutions on the same build machine using a unique service account) are not supported by default by the Scanner for .NET.
```

Reply:
We clarified a bit this directly and we see here two scenarios:
- when `AnalysisContext.EnableConcurrentExecution` is used: this case is happening in the same process and the `SonarLint.xml` should be the same for all the analyzers (it's the same project)
- when a user has a solution with multiple projects and it has manually configured different `SonarLint.xml` files for each project and it enables parallel builds. In this case, if the projects are compiled in parallel, they are compiled by different work processes and theoretically the list of enabled rules is loaded once per process

Reply:
As a babystep, this should be fine for now. In long term, we need to rework how `SonarLint.xml` is used - it's read on different places in a different ways and not cached in `AnalysisContext` as it should be.